internal/mesh/internal/types/proxy_configuration_test.go

```diff
@@ -8,12 +8,25 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	catalogtesthelpers "github.com/hashicorp/consul/internal/catalog/catalogtest/helpers"
+	"github.com/hashicorp/consul/internal/resource"
 	"github.com/hashicorp/consul/internal/resource/resourcetest"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
 	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v2beta1"
 	"github.com/hashicorp/consul/proto/private/prototest"
 	"github.com/hashicorp/consul/sdk/iptables"
 )
 
+func TestProxyConfigurationACLs(t *testing.T) {
+	catalogtesthelpers.RunWorkloadSelectingTypeACLsTests[*pbmesh.ProxyConfiguration](t, pbmesh.ProxyConfigurationType,
+		func(selector *pbcatalog.WorkloadSelector) *pbmesh.ProxyConfiguration {
+			return &pbmesh.ProxyConfiguration{Workloads: selector}
+		},
+		func(registry resource.Registry) {
+			RegisterProxyConfiguration(registry)
+		})
```

Comment:
Nit: This should be equivalent
```suggestion
		RegisterProxyConfiguration,
		)
```